File: scratch/parse.js
import fs from 'fs';

const content = fs.readFileSync('src/pages/Home.tsx', 'utf8');
const lines = content.split('\n');

let stack = [];
lines.forEach((line, index) => {
  const lineNum = index + 1;
  
  // A simple regex to find open and close tags of interest (div, motion.div)
  // Note: We need to handle tags that open/close on the same line separately if we want precision,
  // but for tracing hierarchy, we can look at match sequences.
  
  let matches = [];
  // Find all matches in order of occurrence
  const tagRegex = /<\/?(?:div|motion\.div)\b/g;
  let match;
  while ((match = tagRegex.exec(line)) !== null) {
    matches.push({
      type: match[0],
      lineNum,
      content: line.trim()
    });
  }
  
  matches.forEach(m => {
    if (m.type.startsWith('</')) {
      const tag = m.type.substring(2);
      if (stack.length > 0) {
        const last = stack.pop();
        if (last.tag === tag) {
          console.log(`Match: ${tag} ${last.lineNum} -> ${m.lineNum} | ${last.content}`);
        } else {
          console.log(`Mismatch: close ${tag} at ${m.lineNum} closes ${last.tag} from ${last.lineNum} | ${m.content}`);
        }
      } else {
        console.log(`Unmatched close: ${tag} at ${m.lineNum} | ${m.content}`);
      }
    } else {
      const tag = m.type.substring(1);
      stack.push({
        tag,
        lineNum: m.lineNum,
        content: m.content
      });
    }
  });
});

File: scratch/parse.py
import re

def parse_tags(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()
        
    stack = []
    for idx, line in enumerate(lines):
        line_num = idx + 1
        # Find div open tags
        opens = re.findall(r'<div\b', line)
        closes = re.findall(r'</div\b', line)
        motion_opens = re.findall(r'<motion\.div\b', line)
        motion_closes = re.findall(r'</motion\.div\b', line)
        
        for _ in opens:
            stack.append(('div', line_num, line.strip()))
        for _ in motion_opens:
            stack.append(('motion.div', line_num, line.strip()))
            
        for _ in closes:
            if stack:
                tag, start_line, orig = stack.pop()
                if tag == 'div':
                    print(f"Match: div {start_line} -> {line_num} | {orig}")
                else:
                    print(f"Mismatch close: div at {line_num} closes {tag} from {start_line}")
            else:
                print(f"Unmatched close: div at {line_num}")
                
        for _ in motion_closes:
            if stack:
                tag, start_line, orig = stack.pop()
                if tag == 'motion.div':
                    print(f"Match: motion.div {start_line} -> {line_num} | {orig}")
                else:
                    print(f"Mismatch close: motion.div at {line_num} closes {tag} from {start_line}")
            else:
                print(f"Unmatched close: motion.div at {line_num}")

parse_tags('c:/Users/DELL/OneDrive/Desktop/Tanisha/MedSage 15/src/pages/Home.tsx')
